resolver/addressable_resolver.go

```diff
@@ -184,30 +247,24 @@ func (r *URIResolver) URIFromObjectReference(ctx context.Context, ref *corev1.Ob
 		return nil, fmt.Errorf("failed to get object %s/%s: %w", ref.Namespace, ref.Name, err)
 	}
 
-	// K8s Services are special cased. They can be called, even though they do not satisfy the
-	// Callable interface.
-	if ref.APIVersion == "v1" && ref.Kind == "Service" {
-		url := &apis.URL{
-			Scheme: "http",
-			Host:   network.GetServiceHostname(ref.Name, ref.Namespace),
-			Path:   "",
-		}
-		return url, nil
-	}
```

Comment:
We should keep this special case here and return:
```golang
&duckv1.Addressable{URL: url, CACerts: destinationRef.CACerts}
```
In addition, if `destinationRef.CACerts` is non nil and non empty string, `Scheme` for url should be `https`